ICSharpCode.Decompiler/TypeSystem/MetadataModule.cs

```diff
@@ -626,8 +626,70 @@ IMethod CreateFakeMethod(IType declaringType, string name, MethodSignature<IType
 				parameters.Add(new DefaultParameter(type, ""));
 			}
 			m.Parameters = parameters;
+
+			GuessFakeMethodAccessor(declaringType, name, signature, m, parameters);
+
 			return m;
 		}
+
+		private void GuessFakeMethodAccessor(IType declaringType, string name, MethodSignature<IType> signature, FakeMethod m, List<IParameter> parameters)
+		{
+			var guessedGetter = name.StartsWith("get_", StringComparison.Ordinal);
+			var guessedSetter = name.StartsWith("set_", StringComparison.Ordinal);
+			if (guessedGetter || guessedSetter)
+			{
+				var propertyName = name.Substring(4);
+				m.AccessorKind = guessedGetter ? MethodSemanticsAttributes.Getter :
+								 guessedSetter ? MethodSemanticsAttributes.Setter :
+								 default;
+				m.AccessorOwner = new FakeProperty(Compilation) {
+					Name = propertyName,
+					DeclaringType = declaringType,
+					ReturnType = guessedGetter ? signature.ReturnType
+								: guessedSetter ? parameters.Last().Type
+								: throw new InvalidOperationException(),
+					Getter = guessedGetter ? m : null,
+					Setter = guessedSetter ? m : null,
+					IsIndexer = guessedGetter ? parameters.Count > 0
+							  : guessedSetter ? parameters.Count > 1
+							  : throw new InvalidOperationException(),
+					Parameters = parameters,
+					IsStatic = m.IsStatic,
+				};
+			}
+			else
+			{
+				const string addPrefix = "add_";
+				const string removePrefix = "remove_";
+				const string raisePrefix = "raise_";
+				var guessedAdd = name.StartsWith(addPrefix, StringComparison.Ordinal);
+				var guessedRemove = name.StartsWith(removePrefix, StringComparison.Ordinal);
+				var guessedRaise = name.StartsWith(raisePrefix, StringComparison.Ordinal);
+				if (guessedAdd || guessedRemove || guessedRaise)
+				{
+					m.AccessorOwner = new FakeEvent(Compilation) {
+						Name = guessedAdd ? name.Substring(addPrefix.Length) :
+							   guessedRemove ? name.Substring(removePrefix.Length) :
+							   guessedRaise ? name.Substring(raisePrefix.Length)
+							 : throw new InvalidOperationException(),
+						DeclaringType = declaringType,
+						IsStatic = m.IsStatic,
+						AddAccessor = guessedAdd ? m : null,
+						RemoveAccessor = guessedRemove ? m : null,
+						InvokeAccessor = guessedRaise ? m : null,
+						ReturnType = guessedAdd ? parameters.Single().Type :
+									 guessedRemove ? parameters.Single().Type :
```

Comment:
There's no guarantee that `parameters.Count == 1`; this needs to be validated before attempting to create a fake event.